test/save_assemblies_from_fastas_integration_test.py

```diff
@@ -576,4 +584,171 @@ def test_fail_mismatched_inputs(config, impl, context, scratch):
+        'GCF_000007065.1_ASM706v1_genomic.fna.gz'
+    ]
+
+    # copy 2 assembly files into the data dir
+    for file_name in file_names:
+        shutil.copy(data / file_name, tmp_dir)
+
+    params = {
+        'workspace_id': config['ws_id'],
+        'inputs': [
+            {'file': tmp_dir / 'GCA_002506415.1_ASM250641v1_genomic.fna.gz', 'assembly_name': 'GCA_002506415.1_ASM250641v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000007065.1_ASM706v1_genomic.fna.gz', 'assembly_name': 'GCF_000007065.1_ASM706v1_genomic.fna.gz'}
+        ]
+    }
+
+    var_name = "MAX_THREADS"
+    config['file_config']["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "10e"
+    impl = AssemblyUtil(config['file_config'])
+    with raises(Exception) as got:
+        impl.save_assemblies_from_fastas(context, params)
+    assert_exception_correct(got.value, ValueError(f"Cannot evaluate the string {var_name}"))
+
+    config['file_config']["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "-1"
+    impl = AssemblyUtil(config['file_config'])
+    with raises(Exception) as got:
+        impl.save_assemblies_from_fastas(context, params)
+    assert_exception_correct(got.value, ValueError(f"{var_name} cannot be negative"))
+
+
+def test_generator_overflow(config, scratch):
+    tmp_dir = scratch / ("test_generator_overflow" + str(uuid.uuid4()))
+    os.makedirs(tmp_dir)
+    data = Path('data')
+    file_names = [
+        'GCA_002506415.1_ASM250641v1_genomic.fna.gz',
+        'GCF_000007065.1_ASM706v1_genomic.fna.gz',
+        'GCF_000970165.1_ASM97016v1_genomic.fna.gz',
+        'GCF_000970185.1_ASM97018v1_genomic.fna.gz',
+        'GCF_000970205.1_ASM97020v1_genomic.fna.gz',
+        'GCF_000970245.1_ASM97024v1_genomic.fna.gz'
+    ]
+
+    # copy 6 assembly files into the data dir
+    for file_name in file_names:
+        shutil.copy(data / file_name, tmp_dir)
+
+    params = {
+        'workspace_id': config['ws_id'],
+        'inputs': [
+            {'file': tmp_dir / 'GCA_002506415.1_ASM250641v1_genomic.fna.gz', 'assembly_name': 'GCA_002506415.1_ASM250641v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000007065.1_ASM706v1_genomic.fna.gz', 'assembly_name': 'GCF_000007065.1_ASM706v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970165.1_ASM97016v1_genomic.fna.gz', 'assembly_name': 'GCF_000970165.1_ASM97016v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970185.1_ASM97018v1_genomic.fna.gz', 'assembly_name': 'GCF_000970185.1_ASM97018v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970205.1_ASM97020v1_genomic.fna.gz', 'assembly_name': 'GCF_000970205.1_ASM97020v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970245.1_ASM97024v1_genomic.fna.gz', 'assembly_name': 'GCF_000970245.1_ASM97024v1_genomic.fna.gz'}
+        ]
+    }
+
+    dfu = DataFileUtil(config['callback_url'], token=config['token'])
+    fta = FastaToAssembly(dfu, scratch)
+    fta.import_fasta_mass(params, 1, 10, 38300)
```

Comment:
If import_fasta_mass did nothing this test would still pass, at least do some checks to make sure things worked

Reply:
Updated!

Reply:
:+1: